lib/roo/excelx/sheet_doc.rb

```diff
@@ -125,11 +132,12 @@ def create_cell_from_value(value_type, cell, formula, format, style, hyperlink,
         #       3. formula
         case value_type
         when :shared
-          value = shared_strings.use_html?(cell.content.to_i) ? shared_strings.to_html[cell.content.to_i] : shared_strings[cell.content.to_i]
-          Excelx::Cell.create_cell(:string, value, formula, style, hyperlink, coordinate)
+          cell_content = cell.content.to_i
+          value = shared_strings.use_html?(cell_content) ? shared_strings.to_html[cell_content] : shared_strings[cell_content]
+          Excelx::Cell.cell_class(:string).new(value, formula, style, hyperlink, coordinate)
```

Comment:
Same question about using `cell_class`. Thinking again though, I guess this does encapsulate the logic so if we were to need to change the name of the class we wouldn't have to go hunting.

Reply:
Yes you are right about the reasoning.